pkg/cloud/services/ec2/instances.go

```diff
@@ -195,7 +196,17 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*infrav1.Instance,
 		record.Warnf(scope.AWSMachine, corev1.EventTypeWarning, "FailedGetBootstrapData", err.Error())
 		return nil, err
 	}
-	input.UserData = pointer.StringPtr(userData)
+
+	if scope.AWSMachine.Spec.EncryptedCloudInit != nil && *scope.AWSMachine.Spec.EncryptedCloudInit {
+		ssmService := ssm.NewService(s.scope)
+		encryptedUserData, err := ssmService.EncryptedUserData(scope)
```

Comment:
The field is called `EncryptedCloudInit`, but the variable `encryptedUserData`. IMO we should use one and be consistent